### ISSUE TITLE
Fail to build with g++14

### ISSUE THREAD

**Issue by marillat** [rbuehlma/pvr.zattoo#209]
From https://bugs.debian.org/1075120 updated for Omega
 pvr.zattoo 21.0.2

```
[1/10] Building CXX object CMakeFiles/pvr.zattoo.dir/src/http/Curl.cpp.o
FAILED: CMakeFiles/pvr.zattoo.dir/src/http/Curl.cpp.o 
/usr/lib/ccache/c++ -DADDON_GLOBAL_VERSION_FILESYSTEM_USED -DADDON_GLOBAL_VERSION_GENERAL_USED -DADDON_GLOBAL_VERSION_MAIN_USED -DADDON_INSTANCE_VERSION_PVR_USED -DBUILD_KODI_ADDON -Dpvr_zattoo_EXPORTS -I/usr/include/kodi -g -O2 -ffile-prefix-map=/build/kodi-pvr-zattoo-dmo-21.0.2=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -DZATTOO_VERSION=21.0.2 -DKODI_VERSION=21.0.0 -O2 -g -DNDEBUG -std=c++17 -fPIC -fvisibility=internal   -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -DHAVE_SSE=1 -DHAVE_SSE2=1 -DHAVE_SSE3=1 -DHAVE_SSSE3=1 -DHAVE_SSE4_1=1 -MD -MT CMakeFiles/pvr.zattoo.dir/src/http/Curl.cpp.o -MF CMakeFiles/pvr.zattoo.dir/src/http/Curl.cpp.o.d -o CMakeFiles/pvr.zattoo.dir/src/http/Curl.cpp.o -c /build/kodi-pvr-zattoo-dmo-21.0.2/src/http/Curl.cpp
In file included from /build/kodi-pvr-zattoo-dmo-21.0.2/src/http/../Utils.h:6,
                 from /build/kodi-pvr-zattoo-dmo-21.0.2/src/http/Curl.cpp:4:
/usr/include/rapidjson/document.h: In member function 'rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)':
/usr/include/rapidjson/document.h:319:82: error: assignment of read-only member 'rapidjson::GenericStringRef<CharType>::length'
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~~^~~~~~~~~~~~
[2/10] Building CXX object CMakeFiles/pvr.zattoo.dir/src/Utils.cpp.o
FAILED: CMakeFiles/pvr.zattoo.dir/src/Utils.cpp.o 
/usr/lib/ccache/c++ -DADDON_GLOBAL_VERSION_FILESYSTEM_USED -DADDON_GLOBAL_VERSION_GENERAL_USED -DADDON_GLOBAL_VERSION_MAIN_USED -DADDON_INSTANCE_VERSION_PVR_USED -DBUILD_KODI_ADDON -Dpvr_zattoo_EXPORTS -I/usr/include/kodi -g -O2 -ffile-prefix-map=/build/kodi-pvr-zattoo-dmo-21.0.2=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -DZATTOO_VERSION=21.0.2 -DKODI_VERSION=21.0.0 -O2 -g -DNDEBUG -std=c++17 -fPIC -fvisibility=internal   -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -DHAVE_SSE=1 -DHAVE_SSE2=1 -DHAVE_SSE3=1 -DHAVE_SSSE3=1 -DHAVE_SSE4_1=1 -MD -MT CMakeFiles/pvr.zattoo.dir/src/Utils.cpp.o -MF CMakeFiles/pvr.zattoo.dir/src/Utils.cpp.o.d -o CMakeFiles/pvr.zattoo.dir/src/Utils.cpp.o -c /build/kodi-pvr-zattoo-dmo-21.0.2/src/Utils.cpp
In file included from /build/kodi-pvr-zattoo-dmo-21.0.2/src/Utils.h:6,
                 from /build/kodi-pvr-zattoo-dmo-21.0.2/src/Utils.cpp:1:
/usr/include/rapidjson/document.h: In member function 'rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)':
/usr/include/rapidjson/document.h:319:82: error: assignment of read-only member 'rapidjson::GenericStringRef<CharType>::length'
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~~^~~~~~~~~~~~
[3/10] Building CXX object CMakeFiles/pvr.zattoo.dir/src/http/HttpClient.cpp.o
FAILED: CMakeFiles/pvr.zattoo.dir/src/http/HttpClient.cpp.o 
/usr/lib/ccache/c++ -DADDON_GLOBAL_VERSION_FILESYSTEM_USED -DADDON_GLOBAL_VERSION_GENERAL_USED -DADDON_GLOBAL_VERSION_MAIN_USED -DADDON_INSTANCE_VERSION_PVR_USED -DBUILD_KODI_ADDON -Dpvr_zattoo_EXPORTS -I/usr/include/kodi -g -O2 -ffile-prefix-map=/build/kodi-pvr-zattoo-dmo-21.0.2=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -DZATTOO_VERSION=21.0.2 -DKODI_VERSION=21.0.0 -O2 -g -DNDEBUG -std=c++17 -fPIC -fvisibility=internal   -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -DHAVE_SSE=1 -DHAVE_SSE2=1 -DHAVE_SSE3=1 -DHAVE_SSSE3=1 -DHAVE_SSE4_1=1 -MD -MT CMakeFiles/pvr.zattoo.dir/src/http/HttpClient.cpp.o -MF CMakeFiles/pvr.zattoo.dir/src/http/HttpClient.cpp.o.d -o CMakeFiles/pvr.zattoo.dir/src/http/HttpClient.cpp.o -c /build/kodi-pvr-zattoo-dmo-21.0.2/src/http/HttpClient.cpp
In file included from /build/kodi-pvr-zattoo-dmo-21.0.2/src/http/Cache.h:4,
                 from /build/kodi-pvr-zattoo-dmo-21.0.2/src/http/HttpClient.cpp:2:
/usr/include/rapidjson/document.h: In member function 'rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)':
/usr/include/rapidjson/document.h:319:82: error: assignment of read-only member 'rapidjson::GenericStringRef<CharType>::length'
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~~^~~~~~~~~~~~
[4/10] Building CXX object CMakeFiles/pvr.zattoo.dir/src/http/Cache.cpp.o
FAILED: CMakeFiles/pvr.zattoo.dir/src/http/Cache.cpp.o 
/usr/lib/ccache/c++ -DADDON_GLOBAL_VERSION_FILESYSTEM_USED -DADDON_GLOBAL_VERSION_GENERAL_USED -DADDON_GLOBAL_VERSION_MAIN_USED -DADDON_INSTANCE_VERSION_PVR_USED -DBUILD_KODI_ADDON -Dpvr_zattoo_EXPORTS -I/usr/include/kodi -g -O2 -ffile-prefix-map=/build/kodi-pvr-zattoo-dmo-21.0.2=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -DZATTOO_VERSION=21.0.2 -DKODI_VERSION=21.0.0 -O2 -g -DNDEBUG -std=c++17 -fPIC -fvisibility=internal   -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -DHAVE_SSE=1 -DHAVE_SSE2=1 -DHAVE_SSE3=1 -DHAVE_SSSE3=1 -DHAVE_SSE4_1=1 -MD -MT CMakeFiles/pvr.zattoo.dir/src/http/Cache.cpp.o -MF CMakeFiles/pvr.zattoo.dir/src/http/Cache.cpp.o.d -o CMakeFiles/pvr.zattoo.dir/src/http/Cache.cpp.o -c /build/kodi-pvr-zattoo-dmo-21.0.2/src/http/Cache.cpp
In file included from /build/kodi-pvr-zattoo-dmo-21.0.2/src/http/Cache.h:4,
                 from /build/kodi-pvr-zattoo-dmo-21.0.2/src/http/Cache.cpp:2:
/usr/include/rapidjson/document.h: In member function 'rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)':
/usr/include/rapidjson/document.h:319:82: error: assignment of read-only member 'rapidjson::GenericStringRef<CharType>::length'
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~~^~~~~~~~~~~~
[5/10] Building CXX object CMakeFiles/pvr.zattoo.dir/src/UpdateThread.cpp.o
FAILED: CMakeFiles/pvr.zattoo.dir/src/UpdateThread.cpp.o 
/usr/lib/ccache/c++ -DADDON_GLOBAL_VERSION_FILESYSTEM_USED -DADDON_GLOBAL_VERSION_GENERAL_USED -DADDON_GLOBAL_VERSION_MAIN_USED -DADDON_INSTANCE_VERSION_PVR_USED -DBUILD_KODI_ADDON -Dpvr_zattoo_EXPORTS -I/usr/include/kodi -g -O2 -ffile-prefix-map=/build/kodi-pvr-zattoo-dmo-21.0.2=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -DZATTOO_VERSION=21.0.2 -DKODI_VERSION=21.0.0 -O2 -g -DNDEBUG -std=c++17 -fPIC -fvisibility=internal   -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -DHAVE_SSE=1 -DHAVE_SSE2=1 -DHAVE_SSE3=1 -DHAVE_SSSE3=1 -DHAVE_SSE4_1=1 -MD -MT CMakeFiles/pvr.zattoo.dir/src/UpdateThread.cpp.o -MF CMakeFiles/pvr.zattoo.dir/src/UpdateThread.cpp.o.d -o CMakeFiles/pvr.zattoo.dir/src/UpdateThread.cpp.o -c /build/kodi-pvr-zattoo-dmo-21.0.2/src/UpdateThread.cpp
In file included from /build/kodi-pvr-zattoo-dmo-21.0.2/src/ZatData.h:6,
                 from /build/kodi-pvr-zattoo-dmo-21.0.2/src/UpdateThread.cpp:4:
/usr/include/rapidjson/document.h: In member function 'rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)':
/usr/include/rapidjson/document.h:319:82: error: assignment of read-only member 'rapidjson::GenericStringRef<CharType>::length'
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~~^~~~~~~~~~~~
[6/10] Building CXX object CMakeFiles/pvr.zattoo.dir/src/Session.cpp.o
FAILED: CMakeFiles/pvr.zattoo.dir/src/Session.cpp.o 
/usr/lib/ccache/c++ -DADDON_GLOBAL_VERSION_FILESYSTEM_USED -DADDON_GLOBAL_VERSION_GENERAL_USED -DADDON_GLOBAL_VERSION_MAIN_USED -DADDON_INSTANCE_VERSION_PVR_USED -DBUILD_KODI_ADDON -Dpvr_zattoo_EXPORTS -I/usr/include/kodi -g -O2 -ffile-prefix-map=/build/kodi-pvr-zattoo-dmo-21.0.2=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -DZATTOO_VERSION=21.0.2 -DKODI_VERSION=21.0.0 -O2 -g -DNDEBUG -std=c++17 -fPIC -fvisibility=internal   -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -DHAVE_SSE=1 -DHAVE_SSE2=1 -DHAVE_SSE3=1 -DHAVE_SSSE3=1 -DHAVE_SSE4_1=1 -MD -MT CMakeFiles/pvr.zattoo.dir/src/Session.cpp.o -MF CMakeFiles/pvr.zattoo.dir/src/Session.cpp.o.d -o CMakeFiles/pvr.zattoo.dir/src/Session.cpp.o -c /build/kodi-pvr-zattoo-dmo-21.0.2/src/Session.cpp
In file included from /build/kodi-pvr-zattoo-dmo-21.0.2/src/Utils.h:6,
                 from /build/kodi-pvr-zattoo-dmo-21.0.2/src/Session.h:6,
                 from /build/kodi-pvr-zattoo-dmo-21.0.2/src/Session.cpp:1:
/usr/include/rapidjson/document.h: In member function 'rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)':
/usr/include/rapidjson/document.h:319:82: error: assignment of read-only member 'rapidjson::GenericStringRef<CharType>::length'
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~~^~~~~~~~~~~~
[7/10] Building CXX object CMakeFiles/pvr.zattoo.dir/src/epg/ZattooEpgProvider.cpp.o
FAILED: CMakeFiles/pvr.zattoo.dir/src/epg/ZattooEpgProvider.cpp.o 
/usr/lib/ccache/c++ -DADDON_GLOBAL_VERSION_FILESYSTEM_USED -DADDON_GLOBAL_VERSION_GENERAL_USED -DADDON_GLOBAL_VERSION_MAIN_USED -DADDON_INSTANCE_VERSION_PVR_USED -DBUILD_KODI_ADDON -Dpvr_zattoo_EXPORTS -I/usr/include/kodi -g -O2 -ffile-prefix-map=/build/kodi-pvr-zattoo-dmo-21.0.2=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -DZATTOO_VERSION=21.0.2 -DKODI_VERSION=21.0.0 -O2 -g -DNDEBUG -std=c++17 -fPIC -fvisibility=internal   -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -DHAVE_SSE=1 -DHAVE_SSE2=1 -DHAVE_SSE3=1 -DHAVE_SSSE3=1 -DHAVE_SSE4_1=1 -MD -MT CMakeFiles/pvr.zattoo.dir/src/epg/ZattooEpgProvider.cpp.o -MF CMakeFiles/pvr.zattoo.dir/src/epg/ZattooEpgProvider.cpp.o.d -o CMakeFiles/pvr.zattoo.dir/src/epg/ZattooEpgProvider.cpp.o -c /build/kodi-pvr-zattoo-dmo-21.0.2/src/epg/ZattooEpgProvider.cpp
In file included from /build/kodi-pvr-zattoo-dmo-21.0.2/src/epg/ZattooEpgProvider.cpp:2:
/usr/include/rapidjson/document.h: In member function 'rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)':
/usr/include/rapidjson/document.h:319:82: error: assignment of read-only member 'rapidjson::GenericStringRef<CharType>::length'
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~~^~~~~~~~~~~~
/build/kodi-pvr-zattoo-dmo-21.0.2/src/epg/ZattooEpgProvider.cpp: In member function 'virtual bool ZattooEpgProvider::LoadEPGForChannel(ZatChannel&, time_t, time_t)':
/build/kodi-pvr-zattoo-dmo-21.0.2/src/epg/ZattooEpgProvider.cpp:73:11: warning: unused variable 'uniqueChannelId' [-Wunused-variable]
   73 |       int uniqueChannelId = Utils::GetChannelId(cid.c_str());
      |           ^~~~~~~~~~~~~~~
[8/10] Building CXX object CMakeFiles/pvr.zattoo.dir/src/ZatData.cpp.o
FAILED: CMakeFiles/pvr.zattoo.dir/src/ZatData.cpp.o 
/usr/lib/ccache/c++ -DADDON_GLOBAL_VERSION_FILESYSTEM_USED -DADDON_GLOBAL_VERSION_GENERAL_USED -DADDON_GLOBAL_VERSION_MAIN_USED -DADDON_INSTANCE_VERSION_PVR_USED -DBUILD_KODI_ADDON -Dpvr_zattoo_EXPORTS -I/usr/include/kodi -g -O2 -ffile-prefix-map=/build/kodi-pvr-zattoo-dmo-21.0.2=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -DZATTOO_VERSION=21.0.2 -DKODI_VERSION=21.0.0 -O2 -g -DNDEBUG -std=c++17 -fPIC -fvisibility=internal   -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -DHAVE_SSE=1 -DHAVE_SSE2=1 -DHAVE_SSE3=1 -DHAVE_SSSE3=1 -DHAVE_SSE4_1=1 -MD -MT CMakeFiles/pvr.zattoo.dir/src/ZatData.cpp.o -MF CMakeFiles/pvr.zattoo.dir/src/ZatData.cpp.o.d -o CMakeFiles/pvr.zattoo.dir/src/ZatData.cpp.o -c /build/kodi-pvr-zattoo-dmo-21.0.2/src/ZatData.cpp
In file included from /build/kodi-pvr-zattoo-dmo-21.0.2/src/ZatData.h:6,
                 from /build/kodi-pvr-zattoo-dmo-21.0.2/src/ZatData.cpp:3:
/usr/include/rapidjson/document.h: In member function 'rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)':
/usr/include/rapidjson/document.h:319:82: error: assignment of read-only member 'rapidjson::GenericStringRef<CharType>::length'
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~~^~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

**Comment by rbuehlma** [comment]
Upstream issue in rapidjson needs to be fixed first:

https://github.com/Tencent/rapidjson/issues/2277

**Comment by marillat** [comment]
OK. Thanks.

**Comment by marillat** [comment]
The Debian package has been fixed in rapidjson 1.1.0+dfsg2-7.3
and pvr.zattoo  build fine, so I close this "bug report"
